src/join_variant_summary_with_clinvar_alleles.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+import sys
+import pandas as pd
+
+from parse_clinvar_xml import HEADER
+
+FINAL_HEADER = HEADER + ['gold_stars', 'pathogenic', 'benign', 'conflicted']
+
+
+def join_variant_summary_with_clinvar_alleles(
+        variant_summary_table, clinvar_alleles_table,
+        genome_build_id="GRCh37"):
+    variant_summary = pd.read_csv(variant_summary_table, sep="\t",
+                                  index_col=False, compression="gzip")
+    print "variant_summary raw", variant_summary.shape
+
+    clinvar_alleles = pd.read_csv(clinvar_alleles_table, sep="\t",
+                                  index_col=False, compression="gzip")
+    print "clinvar_alleles raw", clinvar_alleles.shape
+
+    # use lowercase names and replace . with _ in column names:
+    variant_summary = variant_summary.rename(columns=dict(
+        (col, col.lower().replace(".", "_"))
+        for col in variant_summary.columns))
+    # rename first column to allele_id:
+    variant_summary = variant_summary.rename(
+        columns={variant_summary.columns[0]: "allele_id"})
+
+    # extract relevant columns for the correct assembly and
+    # rename clinicalsignificance, reviewstatus:
+    variant_summary = variant_summary[
+        variant_summary['assembly'] == genome_build_id]
+    variant_summary = variant_summary[
+        ['allele_id', 'clinicalsignificance', 'reviewstatus']]
+    variant_summary = variant_summary.rename(
+        columns={'clinicalsignificance': 'clinical_significance',
+                 'reviewstatus': 'review_status'})
+    print "variant_summary after filter", variant_summary.shape
+
+    # remove clinical_significance and review_status from clinvar_alleles:
+    clinvar_alleles = clinvar_alleles.drop(
+        ['clinical_significance', 'review_status'], axis=1)
+
+    # remove rows with multiple allele_ids
```

Comment:
The input file clinvar_alleles is organized around allele_id, it is not possible to have a row with multiple allele_ids.